SQLite için datasource tanımı eklemeleri

suckerman7 opened the pull request:
İlk issue'da yapılması gereken her şeyi yaptım. Kontrol ederseniz memnun olurum.

suckerman7 committed:
SQLite için datasource tanımı eklemeleri

İlk issue'da yapılması gereken her şeyi yaptım.

hyperpostulate commented on src/main/java/org/mertyuksel/restapidemo/RestApiDemoApplication.java:
Datasource'a dair bean tanımını config paketi içerisinde yer alacak DataAccessConfiguration sınıfına alalım.

suckerman7 replied:
Config paketi içinde şimdi sınıf ile aynı adı taşıyan bir .java dosyasına ekledim Bean tanımını.

hyperpostulate replied:
import'lar yok :)

suckerman7 replied:
4 hatayı gerekli importları ekleyerek tek bir hataya indirgedim. Ama hala DataSource Class'ı için "cannot find symbol" hatası veriyor. Araştırsam da hiç işe yaracak bir çözüm maalesef bulamadım.

Yaptıklarımı pushlayıp siz de bir göz atabilirseniz memnun olurum.

EDIT: şimdi de Configuration class'ı yok diye error veriyor. Acaba DataAccessConfiguration sınıfı ile conflict mi yapıyor diyeceğim ondan da emin değilim. Bu olmayan iki şeyi nasıl import edebileceğim hakkında da internette bilgi yok.

suckerman7 replied:
Mesut bey, uygulamayı sonunda local'de ayağa kaldırabildim. Şimdi Furkan bey ile birlikte hallettik.

hyperpostulate commented on src/main/resources/persistence-sqlite.properties:
url tanımında inmemory değil, file based tanım yapalım. dosya yolunu belirtirken de relative path kullanalım.

suckerman7 replied:
Bunu da son commit'imde hallettim. Göz atabilirseniz memnun olurum.

suckerman7 commented:
Bu task'ta daha yapmam gereken ne vardı? Geçen haftaki toplantıdan sonra buraya yorum eklenmeyince ve ayrıca geçen günlerde de sınavlarıma çalıştığım için maalesef unuttum. Bana hatırlatabilirseniz memnun olurm.

suckerman7 committed:
SQLite üzerinde kalmış tasklar tamamlandı

Eğer eksik olan bir şey varsa altına yorum olarak eklersiniz.

suckerman7 commented:
toplantıda dediğiniz eklemeleri hem application.properties hem de persistence.properties dosyalarına ekledim. Son commit'ime göz atarsanız memnun olurum.

Ama bu pull request'in içinde de şu an Maven Action hatası ortaya çıktı. Sanıyorsam bu PR'i main ile mergeledikten sonra hata gidebilir.

suckerman7 committed:
Bean'i class'ın altına taşıdım

suckerman7 committed:
Revert "Bean'i class'ın altına taşıdım"

This reverts commit 2fb461922ec0af303bceb35c4f1e51c2c2931eab.

suckerman7 committed:
Review'da yer alan değişiklikler

- Bean'i Config paketi altında DataAccessConfiguration sınıfı ile aynı adı taşıyan bir .java dosyasına taşıdım.
- persistence-sqlite.properties dosyasındaki url tanımını file-based olarak değiştirdim.

suckerman7 committed:
Maven Action hatası çözümü?

Gelen hatayı araştırdım, böyle bir çözüm gördüm ve ekledim, işe yarayacak mı emin değilim...

suckerman7 committed:
Import'lar eklendi

suckerman7 committed:
Update DataAccessConfiguration.java

suckerman7 committed:
local'de ayağa kaldırma çabaları

suckerman7 committed:
Hata sayısını bir taneye indirgedim

Yaptıklarıma rağmen GitHub actions hala "DataSource" class'ı için "Cannot Find Symbol" hatası veriyor. Maalesef onu araştırmalarıma rağmen çözemedim.

suckerman7 committed:
Update DataAccessConfiguration.java

Kaldırmam gereken kalkmamış, hemen kaldırayım.

suckerman7 committed:
Update DataAccessConfiguration.java

suckerman7 committed:
Update DataAccessConfiguration.java

configuration class'ı böyle tanımlanabilir mi acaba?

suckerman7 committed:
Revert "Update DataAccessConfiguration.java"

This reverts commit ab190fb570b274ebbbcb217afeae8d17871cdb16.
